### ISSUE TITLE
HTML -> JSON (API call)

### ISSUE THREAD

**Issue by jamesturk** [jamesturk/scrapeghost#18]

**Comment by irux** [comment]
I am actually curious if this would work with other kind of models. I always have the idea to try to use Bert for this kind of things, but I think a following instructions model would be needed for a good performance.

**Comment by clarkmcc** [comment]
Yes, I'd love to see this on the new alpaca models. The major problem that I see (not understanding how this prompts OpenAI under the hood) is that successful prompts are much trickier with models like alpaca and llama.

**Comment by walking-octopus** [comment]
Hmm.... Didn't Cohere make their models free to call (albeit with a rate limit)? This could make using this much much more viable for scraping a few small pages.

**Comment by daankortenbach** [comment]
I'd love to see support for OpenAssistent models.

**Comment by jamesturk** [comment]
The groundwork for this is there after some recent refactors, I am hoping to get access to Claude soon as with its 100k token limit it'd be amazing to see how it performs.  Updating the parent issue w/ the status of some other models as well.

**Comment by cpoptic** [comment]
How about adding Falcon 7B and/or 40B LLM model support?

**Comment by Sunishchal** [comment]
@jamesturk I'm very interested in support for Claude 100k. Happy to work on a PR for this if you're welcoming contributors.

**Comment by ishaan-jaff** [comment]
@jamesturk expanded non OpenAI model coverage in this PR #55

**Comment by jamesturk** [comment]
Update to those tracking this: 
In general the approach laid out in PR #55 seems like a great way to go, relying on a well maintained library that abstracts away differences between these models saves this library from needing to reinvent that wheel that others have tackled.  I'd been toying with a lightweight version of the same, but hadn't done research on what else was out there yet.

As noted on PR #55 I don't think it's quite ready to be merged in yet, there are other parts of the code base that assume OpenAI that I'll want to check on.  (I also just heard about `litellm` half an hour ago & want to do a tiny bit of due diligence before adding the dependency 😄 )

**Comment by krrishdholakia** [comment]
any update on this @jamesturk

**Comment by cornpo** [comment]
This works well with oobabooga's OpenAi extension with Mistral7B and Phind34B.

https://github.com/briansunter/logseq-plugin-gpt3-openai
